beacon_node/lighthouse_network/src/peer_manager/peerdb.rs

```diff
@@ -723,6 +723,17 @@ impl<E: EthSpec> PeerDB<E> {
                 .map(|csc| csc.into())
                 .collect();
             peer_info.set_custody_subnets(all_subnets);
+        } else {
+            let peer_info = self.peers.get_mut(&peer_id).expect("peer exists");
+            let node_id = peer_id_to_node_id(&peer_id).expect("convert peer_id to node_id");
+            let subnets = DataColumnSubnetId::compute_custody_subnets::<E>(
+                node_id.raw(),
+                spec.custody_requirement + 4,
```

Comment:
`spec.custody_requirement` did not fix the flakiness; the failure where `cx.get_custodial_peers(self.column_index)` returns only a supernode still occurs. Based on my testing, `spec.custody_requirement + 4` (or more) seems to fix it.

Reply:
Why not have two supernodes in the retry test + `set_custody_subnets` in all tests for completeness?

Reply:
That makes sense to me💡 Thanks!

Reply:
Yeah the random generated peer ids doesn’t guarantee that peers are distributed across all subnets evenly. we need to either depend on more super nodes or have deterministic node ids in the test. In practice, LH will need to continuously maintain peers across all subnets.